File: apps/konekto_portal/lib/auth/staff_session.dart
import 'package:konekto_portal/auth/staff_role.dart';

class StaffSession {
  final String uid;
  final String hotelId;
  final StaffRole role;
  final String name;
  final String email;

  const StaffSession({
    required this.uid,
    required this.hotelId,
    required this.role,
    required this.name,
    required this.email,
  });

  factory StaffSession.fromJson(Map<String, dynamic> json) {
    return StaffSession(
      uid: json['id'] as String,
      hotelId: json['hotelId'] as String,
      role: StaffRole.fromString(json['role'] as String),
      name: json['name'] as String? ?? '',
      email: json['email'] as String? ?? '',
    );
  }
}

File: apps/konekto_portal_next/types/staffSession.ts
/** Portado de apps/konekto_portal/lib/auth/staff_role.dart + staff_session.dart. */
export type StaffRole = 'gerente' | 'recepcao'

export interface StaffSession {
  uid: string
  hotelId: string
  role: StaffRole
  name: string
  email: string
}

interface StaffSessionJson {
  id: string
  hotelId: string
  role: string
  name?: string | null
  email?: string | null
}

export function staffSessionFromJson(json: StaffSessionJson): StaffSession {
  return {
    uid: json.id,
    hotelId: json.hotelId,
    role: json.role as StaffRole,
    name: json.name ?? '',
    email: json.email ?? '',
  }
}

export const staffRoleLabel: Record<StaffRole, string> = {
  gerente: 'Gerente',
  recepcao: 'Recepção',
}
